File: Binary Search/Level_Order_Traversal.py
class Node:
    def __init__(self, key):
        self.data = key
        self.left = None
        self.right = None
def printLevelOrder(root):
    h = height(root)
    for i in range(1, h+1):
        printCurrentLevel(root, i)
def printCurrentLevel(root, level):
    if root is None:
        return
    if level == 1:
        print(root.data, end=" ")
    elif level > 1:
        printCurrentLevel(root.left, level-1)
        printCurrentLevel(root.right, level-1)
def height(node):
    if node is None:
        return 0
    else:
        lheight = height(node.left)
        rheight = height(node.right)
        return max(lheight,rheight) + 1 
if __name__ == '__main__':
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)
    root.left.right = Node(5)
    root.right.left = Node(6)
    root.right.left.left = Node(7)
    printLevelOrder(root)

File: Binary Search/Level_Order_Traversal.java
package BinaryTree;
import java.util.*;
class Node {
    int data;
    Node right, left;
    public Node(int data) {
        this.data = data;
        this.right = null;
        this.left = null;
    }
}
public class LevelOrderTraversal {
    static void LevelOrder(Node root) {
        Queue<Node> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()) {
            Node current = queue.poll();
            System.out.print(current.data + " ");
            if (current.left != null)
                queue.add(current.left);
            if (current.right != null)
                queue.add(current.right);
            } 
        }
    public static void main(String[] args) {
            Node root = new Node(1);
            root.left = new Node(2);
            root.right = new Node(3);
            root.left.right = new Node(4);
            root.left.left = new Node(5);
            LevelOrder(root);
         
        }
    }
